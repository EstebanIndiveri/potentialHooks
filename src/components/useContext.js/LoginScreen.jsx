import React from 'react'

const LoginScreen = () => {

    //1.referenia al context
    //2.setuser
    //3.
/*
{
    id,
    name
}
*/
    return ( 
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button 
            className="btn btn-secondary"
            // onClick={()=>serUser()}
            >

            </button>
        </div>
     );
}
 
export default LoginScreen;