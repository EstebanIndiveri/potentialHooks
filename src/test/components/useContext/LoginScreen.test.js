import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import LoginScreen from '../../../components/useContext.js/LoginScreen';
import { UserContext } from '../../../components/useContext.js/UserContext';

    describe('<LoginScreen/>', () => {
        const user={
            name:'fernan',
            email:'gmai.com'
        }
        const setUser=jest.fn();
        const wrapper=mount(
            <UserContext.Provider value={{user,setUser}}>
                <LoginScreen/>
            </UserContext.Provider>
        );
        
        test('matchsnapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
        test('ejecutar argunmento esperado', () => {
            wrapper.find('button').prop('onClick')();
            expect(setUser).toHaveBeenCalledWith({
                id:123,
                name:"algo"
            });
        });
        
        
    })
    