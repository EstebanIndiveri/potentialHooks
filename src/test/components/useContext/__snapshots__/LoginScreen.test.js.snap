// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen/> matchsnapshot 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
